DevAcademy/Components/CircleButton.swift

```diff
@@ -0,0 +1,33 @@
+//
+//  CircleButton.swift
+//  DevAcademy
+//
+//  Created by Jan Vaverka on 17.08.2023.
+//
+
+import SwiftUI
+
+struct CircleButton: View {
+    let icon: String
+    let url: String
+
+    var body: some View {
+        Button {
+            if let url = URL(string: url) {
```

Comment:
a tady uz bys mohl volat pouze 
```Swift
Button {
    UIApplication.shared.open(url)
} label: {
    ...
}
```

Reply:
Ale kdybys ten button chtel jeste o neco univerzalejsi, tak bys do nej mohl predavat jenom tu akci. Ze by tu akci potom obslouzil az ViewState. Takze by ten button zustal "hloupej" univerzalni a co by se delo po stisku toho buttonu by resil az ViewState.

Ze by to bylo nejak takhle:
```Swift
struct CircleButton: View {
    let icon: String
    let action: () -> Void

    var body: some View {
        Button(action: action) {
            Image(systemName: icon).font(.title)
        }
        .frame(width: 60, height: 60)
        .foregroundColor(Color.white)
        .background(Color.blue)
        .clipShape(Circle())
    }
}
```